### PR TITLE
MEPTS-346: Change Request in Monthly Report (Resumo Mensal)

### PR DESCRIPTION
**Pull request by ynurmahomed**

### CODE REVIEW

File: api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/ResumoMensalCohortQueries.java [hunk 1]
```diff
@@ -187,7 +187,8 @@ public CohortDefinition getPatientsWhoInitiatedTarvAtThisFacilityDuringCurrentMo
    */
   public CohortDefinition
       getNumberOfPatientsTransferredInFromOtherHealthFacilitiesDuringCurrentMonthB2() {
-
+    // TODO maybe we should be re-using HivCohortQueries#getPatientsTransferredFromOtherHealthFacility
```
Comment (ningosi) [inline api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/ResumoMensalCohortQueries.java:190]: Did you get any response from the BA

Comment (ynurmahomed) [reply api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/ResumoMensalCohortQueries.java:190]: I got no response.

File: api/src/main/java/org/openmrs/module/eptsreports/reporting/library/queries/ResumoMensalQueries.java [hunk 2]
```diff
@@ -63,24 +69,40 @@ public static String getPatientsTransferredFromAnotherHealthFacilityDuringTheCur
             + "         ON type.encounter_id = e.encounter_id "
             + "WHERE  p.voided = 0 "
             + "       AND e.voided = 0 "
-            + "       AND e.encounter_type = %d "
+            + "       AND e.encounter_type = ${masterCard} "
```
Comment (ningosi) [inline api/src/main/java/org/openmrs/module/eptsreports/reporting/library/queries/ResumoMensalQueries.java:72]: What does this phrase means in the specs? **Select all patients with the earliest ART Start Date from the following criterias** Do you need to select all those dates per patient, then later take the first one(one that came first)/

Comment (ynurmahomed) [reply api/src/main/java/org/openmrs/module/eptsreports/reporting/library/queries/ResumoMensalQueries.java:72]: I think it is, as implemented in `InitialArtStartDateCalculation`.

File: api/src/main/java/org/openmrs/module/eptsreports/reporting/library/queries/ResumoMensalQueries.java [hunk 3]
```diff
@@ -63,24 +69,40 @@ public static String getPatientsTransferredFromAnotherHealthFacilityDuringTheCur
...
             + "       AND e.location_id = :location "
             + "       AND e.encounter_datetime BETWEEN :startDate AND :endDate "
             + "       AND transf.voided = 0 "
-            + "       AND transf.concept_id = %d "
-            + "       AND transf.value_coded = %d "
+            + "       AND transf.concept_id = ${transferFromOther} "
```
Comment (ningosi) [inline api/src/main/java/org/openmrs/module/eptsreports/reporting/library/queries/ResumoMensalQueries.java:76]: The specs says **Exclude all patients transferred-in** based on those parameters, looks like we are getting sum of those patients in addition to the lower union. Are we NOt excluding those from the resultant query?

Comment (ynurmahomed) [reply api/src/main/java/org/openmrs/module/eptsreports/reporting/library/queries/ResumoMensalQueries.java:76]: @ningosi I think you might be looking at specs for B1, this is B2.

File: api/src/main/java/org/openmrs/module/eptsreports/reporting/library/queries/ResumoMensalQueries.java (hunk 2 above)
Comment (ningosi) [reply api/src/main/java/org/openmrs/module/eptsreports/reporting/library/queries/ResumoMensalQueries.java:72]: in that class tarnslated to sql would be **UNION of all those subqueries(by getting patient_id and dateRequired, duplicates will be allowed here), then later have an outer query that  select patientId with MIN(dateRequired), here unique rows will be picked, then fhave an outer query that  pick a set of patientIDs, That is how I would translate that class to sql**

**Review by ningosi — APPROVED** [review]
The change matches what B2 needs. Thanks for clarification @ynurmahomed

**Pull request merged**